src/main/java/io/meeds/qa/ui/pages/PeoplePage.java

```diff
@@ -393,5 +393,30 @@ public boolean searchUser(String user) {
     } while (!visible && retry++ < MAX_WAIT_RETRIES); // NOSONAR
     return false;
   }
+  public void searchUserProfile(String user) {
+    searchPeopleInputElement().setTextValue(user);
+  }
+
+  public Boolean userProfileIsDisplayed(String user) {
+    boolean visible = false;
+    int retry = 0;
+    Duration retryWaitTime = Duration.ofSeconds(1);
+    ElementFacade userCard;
+    do {
+      waitFor(300).milliseconds();
+      waitForLoading();
+      userCard = getUserButton(user);
+      userCard.setImplicitTimeout(retryWaitTime);
+      visible = userCard.isVisible();
+      if (visible) {
+        return true;
+      }
+    } while (!visible && retry++ < MAX_WAIT_RETRIES); // NOSONAR
+    return false;
+  }
+
+  public void checkThatTheSearchedUserProfileIsDisplayed(String user) {
+    assertThat(userProfileIsDisplayed(user)).isTrue().as("User " + user + " wasn't found");
```

Comment:
you dont need to redefine the wait. All this stuff exists.

You need to target the elementFacade you search, for example a div with the class peopleCardBody, containing the name of the searched user,
and then with elementFacade.waitUntilVisible, it will automatically wait until the element is visible, and if it is not, if will fail the test.
You have also function like waitLoading for example if you need to wait the end of the search users.

Reply:
done!